doc/trajopt_planner/src/trajopt_example.cpp

```diff
@@ -0,0 +1,269 @@
+#include <ros/ros.h>
+#include <ros/console.h>
+#include <moveit/planning_interface/planning_interface.h>
+
+#include <moveit/move_group_interface/move_group_interface.h>
+#include <moveit/planning_scene_interface/planning_scene_interface.h>
+
+#include <moveit/robot_model_loader/robot_model_loader.h>
+#include <moveit/planning_scene/planning_scene.h>
+#include <moveit/robot_state/robot_state.h>
+
+#include <moveit/planning_scene_monitor/planning_scene_monitor.h>
+#include <moveit/kinematic_constraints/utils.h>
+#include <moveit_visual_tools/moveit_visual_tools.h>
+#include "moveit/planning_interface/planning_request.h"
+#include <moveit/robot_model/robot_model.h>
+#include <moveit/robot_state/robot_state.h>
+
+#include <moveit_msgs/DisplayTrajectory.h>
+#include <moveit_msgs/PlanningScene.h>
+
+#include <geometry_msgs/TransformStamped.h>
+#include <tf2/utils.h>
+#include <tf2_eigen/tf2_eigen.h>
+#include <tf2_ros/transform_listener.h>
+
+/* Author: Omid Heidari
+   Desc: This file is a test for using trajopt in MoveIt. The goal is to make different types of constraints in
+   MotionPlanRequest and visualize the result calculated by using trajopt planner.
+*/
+
+int main(int argc, char** argv)
+{
+  const std::string NODE_NAME = "trajopt_example";
+  ros::init(argc, argv, NODE_NAME);
+  ros::AsyncSpinner spinner(1);
+  spinner.start();
+  ros::NodeHandle node_handle("~");
+
+  const std::string PLANNING_GROUP = "panda_arm";
+  const std::string ROBOT_DESCRIPTION = "robot_description";
+  robot_model_loader::RobotModelLoaderPtr robot_model_loader(
+      new robot_model_loader::RobotModelLoader(ROBOT_DESCRIPTION));
+  robot_model::RobotModelPtr robot_model = robot_model_loader->getModel();
+
+  // Create a RobotState and JointModelGroup to keep track of the current robot pose and planning group
+  robot_state::RobotStatePtr current_state(new robot_state::RobotState(robot_model));
+  current_state->setToDefaultValues();
+
+  const robot_state::JointModelGroup* joint_model_group = current_state->getJointModelGroup(PLANNING_GROUP);
+  const std::vector<std::string>& joint_names = joint_model_group->getActiveJointModelNames();
+
+  // With the planning scene we create a planing scene monitor
+  planning_scene_monitor::PlanningSceneMonitorPtr psm(new planning_scene_monitor::PlanningSceneMonitor(ROBOT_DESCRIPTION));
+  psm->startPublishingPlanningScene(planning_scene_monitor::PlanningSceneMonitor::UPDATE_SCENE);
+  psm->startStateMonitor();
+  psm->startSceneMonitor();
```

Comment:
Duplicate `startSceneMonitor()`

Reply:
the first one is startStateMonitor if you mean line 57 is a duplicate of line 56?

Reply:
Sorry, read too fast and `State` and `Scene` looked the same.